### PR TITLE
Creating a tutorial presenting the abstract representation of a sequence, adding json.dumps options for the serialization

### LINKED ISSUE #498: [DOC] Abstract representation
Hello I was looking in the docs for some paragraph about the abstract representation of a sequence but couldn't find any mention of it; maybe I missed it. There is this [section](https://pulser.readthedocs.io/en/stable/tutorials/serialization.html) about serialization but it uses the "python representation". 

I thought we wanted to encourage as much as possible using the abstract representation so it could be nice to have a small description in there!

### PR DESCRIPTION
**Pull request by a-corni**
I have added an option `json_dumps_options` in `Sequence.to_abstract_repr` to define a mapping between optional keywords of `json.dumps` and its values. 
Closes #498

### CODE REVIEW

File: tutorials/advanced_features/Serialization.ipynb [hunk 1]
```diff
@@ -9,10 +9,11 @@
   },
```
Comment (HGSilveri) [inline tutorials/advanced_features/Serialization.ipynb:9]: <p>It's not only for the human-readability, there are better arguments but they're not really relevant to get into here. What's most important is to stress that the abstract representation should always be the preferred way to serialize a sequence (even if readability is not relevant).</p><p>Also, you are not showing only 180 characters</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pasqal-io/Pulser/pull/515/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

File: tutorials/advanced_features/Serialization.ipynb [hunk 2]
```diff
@@ -9,15 +9,15 @@
...
```
Comment (HGSilveri) [inline tutorials/advanced_features/Serialization.ipynb:9]: <p>*as will be seen...</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pasqal-io/Pulser/pull/515/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

File: tutorials/advanced_features/Serialization.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (a-corni) [reply tutorials/advanced_features/Serialization.ipynb:9]: <p>Indeed ^^'</p>

File: tutorials/advanced_features/Serialization.ipynb [hunk 3]
```diff
@@ -14,10 +14,10 @@
    "outputs": [],
```
[the remark this answers is not in the document]
Comment (a-corni) [reply tutorials/advanced_features/Serialization.ipynb:14]: <p>It has had a reason but now it should not be there ^^`</p><p> </p>

**Review by HGSilveri — APPROVED** [review]
I don't know why but there's an empty code cell at the end of the notebook. LGTM though, feel free to merge when you're ready (make sure to _Squash and Merge_!)

**Pull request merged**